### PR TITLE
feat: add support for Gradle

### PR DESCRIPTION
**Pull request by stuartwdouglas**

### CODE REVIEW

File: internal/moduleconfig/moduleconfig.go [hunk 1]
```diff
@@ -130,21 +135,40 @@ func setConfigDefaults(moduleDir string, config *ModuleConfig) error {
...
+			if config.Java.BuildTool == JavaBuildToolMaven || fileExists(pom) {
+				config.Java.BuildTool = JavaBuildToolMaven
+				config.Build = "mvn -B package"
+				if config.DeployDir == "" {
+					config.DeployDir = "target"
+				}
+				if len(config.Watch) == 0 {
+					config.Watch = []string{"pom.xml", "src/**", "target/generated-sources"}
+				}
+			} else if config.Java.BuildTool == JavaBuildToolGradle || fileExists(buildGradle) || fileExists(buildGradleKts) {
```
Comment (alecthomas) [inline internal/moduleconfig/moduleconfig.go:152]: Nice.

File: internal/moduleconfig/moduleconfig.go [hunk 2]
```diff
@@ -178,17 +202,28 @@ func setConfigDefaults(moduleDir string, config *ModuleConfig) error {
...
 
+func fileExists(filename string) bool {
+	_, err := os.Stat(filename)
+	if err == nil {
+		return true
+	}
+	if os.IsNotExist(err) {
+		return false
+	}
+	return false
```
Comment (alecthomas) [inline internal/moduleconfig/moduleconfig.go:224]: This should be equivalent:
```suggestion
	return err == nil
```

File: jvm-runtime/testdata/java/gradle/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,36 @@
+plugins {
```
Comment (alecthomas) [inline jvm-runtime/testdata/java/gradle/build.gradle.kts:1]: Noooooooooooo

**Review by alecthomas — APPROVED** [review]

File: internal/moduleconfig/moduleconfig.go (hunk 2 above)
Comment (stuartwdouglas) [reply internal/moduleconfig/moduleconfig.go:224]: Oops, I may have just copied that method from copilot.

**Pull request merged**